filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -2167,6 +2167,23 @@ void OpenGLDriver::beginRenderPass(Handle<HwRenderTarget> rth,
         }
     }
 
+    if (rt->gl.fbo_read != rt->gl.fbo) {
+        // we have a multi-sample RenderTarget with non multi-sample attachments (i.e. this is the
+        // EXT_multisampled_render_to_texture emulation).
+        // We need to perform a "backward" resolve, i.e. load the resolved texture into the tile,
+        // everything must appear as though the multi-sample buffer was lost.
+        // We only copy the non msaa buffers that were not discarded or cleared.
+        const TargetBufferFlags resolve = TargetBufferFlags(rt->gl.resolve & ~(discardFlags|clearFlags));
+        GLbitfield mask = getDiscardBits(TargetBufferFlags(resolve));
```

Comment:
Minor nits: (1) I think there's no need for the cast (2) `getDiscardBits` is a fairly generic function so I would've named it `getAttachmentMask` or something like that.

Reply:
Funny you should mention the cast -- this is driving me nuts too. PR coming that makes this much better.